configuration.py

```diff
@@ -0,0 +1,153 @@
+import yaml
+import datetime
+import calendar
+import re
+from sharepoint_utils import print_and_log
+
+
+class Configuration:
```

Comment:
It might be overkill right now, but as Crest develops more connectors, I'd suggest that a lot of the common stuff here (Workplace Search configurations, logging, yaml parsing, checkpointing, etc) move into its own, shared repo/wheel.

Reply:
That's a nice suggestion. I'll try to implement that.